From 50ad8895b6bb3b3437105c4cc433d3a2aa3e2547 Mon Sep 17 00:00:00 2001
From: Jouni Malinen <jouni@qca.qualcomm.com>
Date: Thu, 1 Mar 2012 01:16:06 +0200
Subject: [PATCH 53/73] P2P: Fix p2p_cancel to return success if GO
 Negotiation is stopped

If a GO Negotiation peer is found, wpas_p2p_stop_find() stops the
negotiation and p2p_cancel can return success.

Signed-hostap: Jouni Malinen <jouni@qca.qualcomm.com>
(cherry picked from commit 75c208b9db7ec027ea923ff8af2b408da110bcc2)
---
 wpa_supplicant/p2p_supplicant.c |    1 +
 1 file changed, 1 insertion(+)

--- a/wpa_supplicant/p2p_supplicant.c
+++ b/wpa_supplicant/p2p_supplicant.c
@@ -4288,6 +4288,7 @@ int wpas_p2p_cancel(struct wpa_supplican
 		wpa_printf(MSG_DEBUG, "P2P: Unauthorize pending GO Neg peer "
 			   MACSTR, MAC2STR(peer));
 		p2p_unauthorize(global->p2p, peer);
+		found = 1;
 	}
 
 	wpas_p2p_stop_find(wpa_s);
